pkg/registry/apis/dashboard/register.go

```diff
@@ -144,42 +183,365 @@ func (b *DashboardsAPIBuilder) InstallSchema(scheme *runtime.Scheme) error {
+	}
+
+	// Validate generation conflicts
+	if oldAccessor.GetGeneration() > newAccessor.GetGeneration() {
+		return dashboards.ErrDashboardVersionMismatch // TODO: Check if this is the correct error to return
+	}
+
 	return nil
 }
 
+// validateBasicProperties validates basic dashboard properties
+func (b *DashboardsAPIBuilder) validateBasicProperties(title string, accessor utils.GrafanaMetaAccessor) error {
+	// Validate title
+	if title == "" {
+		return fmt.Errorf("dashboard title cannot be empty")
+	}
+
+	if len(title) > 5000 {
+		return fmt.Errorf("dashboard title is too long (max 5000 characters)")
+	}
+
+	// Validate UID - get it from accesso
+	uid := accessor.GetName()
+	if uid != "" {
+		// Check UID length
+		if len(uid) > 40 { // TODO: Set correct length
+			return fmt.Errorf("dashboard UID is too long (max 40 characters)")
+		}
+
+		// Check valid UID format using regex
+		uidPattern := regexp.MustCompile(`^[a-zA-Z0-9\-\_]+$`) // TODO: Make const outside
+		if !uidPattern.MatchString(uid) {
+			return fmt.Errorf("dashboard UID can only contain alphanumeric characters, dashes and underscores")
+		}
+	}
+
+	// Validate message
+	if message := accessor.GetMessage(); message != "" && len(message) > 500 {
+		return fmt.Errorf("dashboard update message is too long (max 500 characters)")
+	}
+
+	return nil
+}
+
+// validateFolderExists checks if a folder exists
+func (b *DashboardsAPIBuilder) validateFolderExists(ctx context.Context, folderUID string, orgID int64) error {
+	// Check if folder exists using the folder store
+	_, err := b.folderClient.Get(ctx, folderUID, orgID, metav1.GetOptions{})
+
+	if err != nil {
+		if errors.Is(err, dashboards.ErrFolderNotFound) {
+			return err
+		}
+		return fmt.Errorf("error checking folder existence: %w", err)
+	}
+
+	return nil
+}
+
+// validateRefreshInterval validates dashboard refresh interval
+func (b *DashboardsAPIBuilder) validateRefreshInterval(refresh string) error {
+	if refresh == "" || refresh == "auto" {
+		return nil
+	}
+
+	// Skip validation for disabled refresh
+	if refresh == "0" || strings.ToLower(refresh) == "off" {
+		return nil
+	}
+
+	minRefreshInterval := b.cfg.MinRefreshInterval
+	if minRefreshInterval == "" {
+		return nil
+	}
+
+	minInterval, err := time.ParseDuration(minRefreshInterval)
+	if err != nil {
+		return fmt.Errorf("invalid min refresh interval format: %s", minRefreshInterval)
+	}
+
+	refreshInterval, err := time.ParseDuration(refresh)
+	if err != nil {
+		return fmt.Errorf("invalid refresh interval format: %s", refresh)
+	}
+
+	if refreshInterval < minInterval {
+		return dashboards.ErrDashboardRefreshIntervalTooShort
+	}
+
+	return nil
+}
+
+// validateProvisionedDashboardUpdate checks if a provisioned dashboard can be updated
+func (b *DashboardsAPIBuilder) validateProvisionedDashboardUpdate(ctx context.Context, meta utils.GrafanaMetaAccessor) error {
+	manager, ok := meta.GetManagerProperties()
+	if !ok {
+		return nil
+	}
+
+	if manager.Kind == "" {
```

Comment:
NIT (in case we keep this code):
```suggestion
	if manager.Kind == utis.ManagerKindUnknown {
```